app/Http/Controllers/TagController.php

```diff
@@ -38,6 +41,20 @@ public function list(ListTagRequest $request): Collection
 		));
 	}
 
+	public function edit(EditTagRequest $request): void
+	{
+		$tag = $request->tag();
+
+		if (Tag::where('name', $request->name())->exists()) {
+			throw new TagAlreadyExistException();
+		}
```

Comment:
Could this be left to the validator? https://laravel.com/docs/10.x/validation#rule-unique

Reply:
@JasonMillward this is refactored heavily in #3599 and thus no longer applicable. :)

Reply:
<img width="1597" height="683" alt="image" src="https://github.com/user-attachments/assets/b00b92f7-b80e-48ac-a97e-0a151a1d56f4" />
<img width="1365" height="1665" alt="image" src="https://github.com/user-attachments/assets/19a69e48-1409-41d0-9759-b1d07db287b6" />